tests/unit/test_charm.py

```diff
@@ -124,6 +125,43 @@ def test_given_amf_charm_in_active_state_when_nrf_relation_breaks_then_status_is
             BlockedStatus("Waiting for fiveg-nrf relation"),
         )
 
+    @patch("ops.model.Container.pull")
+    @patch("ops.model.Container.exists")
+    @patch("ops.model.Container.push", new=Mock)
+    @patch("charm.check_output")
+    @patch("charms.sdcore_nrf.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
+    @patch("charms.data_platform_libs.v0.data_interfaces.DatabaseRequires.is_resource_created")
+    def test_given_amf_charm_in_active_state_when_database_relation_breaks_then_status_is_blocked(
+        self,
+        patch_is_resource_created,
+        patch_nrf_url,
+        patch_check_output,
+        patch_exists,
+        patch_pull,
+    ):
+        patch_pull.return_value = StringIO(
+            self._read_file("tests/unit/expected_config/config.conf").strip()
+        )
+        patch_exists.return_value = True
+        patch_check_output.return_value = b"1.1.1.1"
+        patch_exists.return_value = True
+        patch_is_resource_created.return_value = True
+        patch_nrf_url.return_value = "http://nrf:8081"
+        self.harness.set_can_connect(container="amf", val=True)
+        self.harness.add_relation(relation_name="fiveg-nrf", remote_app="nrf")
+        database_relation_id = self._database_is_available()
+        self.harness.add_relation(
+            relation_name="certificates", remote_app="tls-certificates-operator"
+        )
+        self.harness.container_pebble_ready("amf")
+
+        self.harness.remove_relation(database_relation_id)
+
+        self.assertEqual(
+            self.harness.model.unit.status,
```

Comment:
We can reduce Container.pull and push mocks using new harness.add_storage feature. Here there is a sample PR. 
https://github.com/canonical/vault-k8s-operator/pull/59/files. This is also correct but could be improved by using new ops features.

Reply:
Since we already have a ticket for that, I would keep it separate and handle that for all tests at once.